### PR TITLE
T0824 - Correção no sistema de parcelas

### PR DESCRIPTION
**Pull request by mstuttgart**
# Description

Correção do sistema de faturamento para novo financeiro

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Criar fatura com parcelas e confirmar a mesma. Deve se criar uma account.move para parcela.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by rodrigets** [comment]
@mstuttgart por gentileza, verificar o build do runbot e ver o que houve de errado.

**Comment by mstuttgart** [comment]
@rodrigets eu não consigo acessar o runbot fora da rede da multi :smile: Mas logo que chegar ai vou ajeitar isso!

**Comment by rodrigets** [comment]
@mstuttgart , rsrs beleza.
Se quiser adiantar alguma coisa, to colocando a imagem do log.
image

'''                      module l10n_generic_coa: an exception occurred in a test
Traceback (most recent call last):
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/modules/loading.py", line 40, in load_test
    _load_data(cr, module_name, idref, mode, 'test')
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/modules/loading.py", line 94, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/tools/convert.py", line 789, in convert_file
    convert_yaml_import(cr, module, fp, kind, idref, mode, noupdate, report)
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/tools/yaml_import.py", line 854, in yaml_import
    yaml_interpreter.process(yaml_string)
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/tools/yaml_import.py", line 797, in process
    self._process_node(node)
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/tools/yaml_import.py", line 810, in _process_node
    self.process_python(node)
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/tools/yaml_import.py", line 601, in process_python
    unsafe_eval(code_obj, {'ref': self.get_id}, code_context)
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/addons/account/test/account_invoice_state.yml", line 48, in <module>
    self.action_invoice_cancel()
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/addons/account/models/account_invoice.py", line 802, in action_invoice_cancel
    return self.action_cancel()
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/addons/br_account_einvoice/models/account_invoice.py", line 231, in action_cancel
    res = super(AccountInvoice, self).action_cancel()
  File "/opt/runbot/multierp/src/others/odoo-extra/runbot/static/build/01999-11-0-T0824-ed5c3a/odoo/addons/br_account/models/account_invoice.py", line 701, in action_cancel
    return super(AccountInvoice, self).account_cancel()
AttributeError: 'super' object has no attribute 'account_cancel' '''

File: br_account/models/account_invoice.py [hunk 1]
```diff
@@ -478,54 +466,101 @@ def action_move_create(self):
...
+                    'move_name': move.name,
+                }
 
-            inv.with_context(ctx).write(values)
+                inv.with_context(ctx).write(values)
 
         return True
 
+    def _compute_residual(self):
+
```
Comment (aldo774) [inline br_account/models/account_invoice.py]: @mstuttgart lembro que você chegou a comentar sobre esse método, mesmo que tenha tentado adaptar o comportamento utilizado previamente nesse método, acredito que você consiga deixa-lo melhor, ta bem pouco coeso, tenta colocar alguns comentários... vi que implementou um teste abaixo pra confirmar a fatura, esta implementação esta sendo testada junto ?

Comment (mstuttgart) [reply br_account/models/account_invoice.py]: @aldo774 realmente, faltou escrever teste para esse método! Obrigado por apontar isso! :+1: 
Eu apenas copiei e adaptei o código do core. Vou dar uma trabalhada nele para deixá-lo mais organizado e quem sabe separar em métodos menores.

**Comment by mstuttgart** [comment]
@aldo774 adicionei teste para o metodo `_compute_residual` e também adicionei testes para o cancelamento da fatura.

**Pull request closed without merging**